### PR TITLE
Add generation of smaller UML diagrams

### LINKED ISSUE #89: Customize UML Diagram documentation to display classes of every namespace separately

### PR DESCRIPTION
**Pull request by DavidRisch**
Fix #89
Until [an issue](https://github.com/thibaultmarin/hpp2plantuml/issues/13) in hpp2plantuml, this only works for some directories not affected by that bug.

### CODE REVIEW

File: doc/uml/build.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    environment = os.environ.copy()
+    environment['PLANTUML_LIMIT_SIZE'] = str(2 ** 14)
+
+    process = subprocess.run([
+        'java',
+        '-jar', jar_path,
+        '--verbose', uml_directory.get_puml_path()
+    ], env=environment)
+
+    if process.returncode != 0:
```
Comment (AnneRei) [inline doc/uml/build.py:93]: use ` process.check_returncode()`

File: doc/uml/build.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        raise RuntimeError()
+
+
+def main():
+    src_directory = os.path.join(os.path.dirname(__file__), '..', '..', 'src')
+    src_directory = os.path.realpath(src_directory)
+
+    for absolute_path, child_directories, child_files in os.walk(src_directory):
+        print(absolute_path)
```
Comment (AnneRei) [inline doc/uml/build.py:102]: Optional: Add a description like "Current directory: " for prettier output

File: doc/uml/build.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+def run_plant_uml(uml_directory: UmlDirectory):
+    current_directory = os.path.realpath(os.path.dirname(__file__))
+
+    jar_path = os.path.join(current_directory, 'bin', 'plantuml.jar')
+
+    environment = os.environ.copy()
+    environment['PLANTUML_LIMIT_SIZE'] = str(2 ** 14)
+
+    process = subprocess.run([
```
Comment (AnneRei) [inline doc/uml/build.py:87]: see subprocess call above

Comment (DavidRisch) [reply doc/uml/build.py:95]: Please clarify

File: doc/uml/build.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    arguments = [
+        '-o', uml_directory.get_puml_path(),
+        '-t', os.path.join(current_directory, 'plantuml_template.puml'),
+    ]
+
+    for child_path in uml_directory.all_contained_files:
+        if child_path.endswith('.h'):
+            arguments += ['-i', child_path]
+
+    process = subprocess.run(['hpp2plantuml'] + arguments)
```
[the remark this answers is not in the document]
Comment (DavidRisch) [reply doc/uml/build.py:77]: > this line throws KeyErrors

This line does not throw any exceptions, the exception is thrown is a separate process.

I don't think adding unnecessary code to improve the output resulting from a temporary 3rd party bug is worth the effort. How would that be 'reused after the issue has been resolved' ?

File: doc/uml/build.py (hunk 3 above)
Comment (AnneRei) [reply doc/uml/build.py:95]: I referred to line 73, but it has been resolved.

File: doc/uml/build.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (AnneRei) [reply doc/uml/build.py:77]: You're right, let's leave it like this.

**Pull request merged**